### PR TITLE
Pipeline, Andrey Karnauhov - 21FPL1

### PR DESCRIPTION
**Pull request by AndreyKarnauhov**

### CODE REVIEW

**Review by demid5111 — CHANGES_REQUESTED** [review]
nothing to review regarding the pipeline, therefore it seems to become a rejected pipeline: there will be no review until the deadline. but you still can pretend to pass with a penalty (-2) next Friday!

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -18,21 +32,46 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+            raise FileNotFoundError
+        if not self.path_to_raw_text_data.is_dir():
+            raise NotADirectoryError
+        if not len(list(self.path_to_raw_text_data.glob('*.*'))):
+            raise EmptyDirectoryError
+        raw_paths = list(self.path_to_raw_text_data.glob('*_raw.txt'))
+        meta_paths = list(self.path_to_raw_text_data.glob('*_meta.json'))
+        if len(raw_paths) != len(meta_paths):
+            raise InconsistentDatasetError
+        for i in range(1, len(raw_paths) + 1):
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:55]: too straightforward and brutal, you can do it better

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -75,6 +115,10 @@ def get_cleaned(self) -> str:
         """
         Returns lowercase original form of a token
         """
+        clean = self._text.lower()
+        for i in [',', '.', '!', '?', ':', '"', '-', '–', ';', '…', '»', '«', '—', '(', ')', '/', '•', '+', '*', '№']:
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: you could reuse string.punctuation

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -96,6 +143,10 @@ def get_cleaned_sentence(self) -> str:
         """
         Returns the lowercase representation of the sentence
         """
+        clean = self._text.lower()
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: Conllu sentence should be built from tokens that it contains. text is needed only for dumping to .conllu file

File: lab_6_pipeline/pos_frequency_pipeline.py [hunk 4]
```diff
@@ -2,6 +2,13 @@
 Implementation of POSFrequencyPipeline for score ten only.
 """
 from typing import Optional
+from pathlib import Path
+from core_utils.article.article import Article
+from lab_6_pipeline.pipeline import ConlluToken, CorpusManager
+
+
+class EmptyFileError(Exception):
+    """."""
```
Comment (demid5111) [inline lab_6_pipeline/pos_frequency_pipeline.py]: ?

**Review by demid5111 — CHANGES_REQUESTED** [review]
nothing special to review. keep going, try to make bigger progress by next review session on Monday

**Comment by demid5111** [comment]
Nothing to review as of today and many comments have not yet been addressed. See you at the oral defense

File: lab_6_pipeline/pipeline.py (hunk 1 above)
Comment (AndreyKarnauhov) [reply lab_6_pipeline/pipeline.py:55]: I know I can do it better, and this knowledge really makes me proud of myself. Not going to implement it though.

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (AndreyKarnauhov) [reply lab_6_pipeline/pipeline.py]: Done

File: lab_6_pipeline/pipeline.py (hunk 3 above)
Comment (AndreyKarnauhov) [reply lab_6_pipeline/pipeline.py]: Fixed

File: lab_6_pipeline/pos_frequency_pipeline.py (hunk 4 above)
Comment (AndreyKarnauhov) [reply lab_6_pipeline/pos_frequency_pipeline.py]: Fixed

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -3,9 +3,24 @@
...
-from core_utils.article.article import SentenceProtocol
+from core_utils.article.article import SentenceProtocol, split_by_sentence
+from core_utils.article.io import from_raw, to_cleaned
 from core_utils.article.ud import OpencorporaTagProtocol, TagConverter
+from core_utils.constants import ASSETS_PATH
+
+
+class EmptyDirectoryError(Exception):
+    """
+    Daaaaamn yo directory's emptie, maaaan
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: .

File: lab_6_pipeline/pipeline.py [hunk 6]
```diff
@@ -3,9 +3,24 @@
...
+    """
+
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:18]: conllutoken and conllusentence - what sort of relationship do they have? how is it called from OOP perspective?

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**